astropy/table/table.py

```diff
@@ -896,24 +900,30 @@ def show_in_notebook(self, tableid=None, css=None, display_length=50,
                                             np.random.randint(1, 1e6))
 
         jsv = JSViewer(display_length=display_length)
-        html = self._base_repr_(html=True, max_width=-1, tableid=tableid,
-                                max_lines=-1, show_dtype=False,
-                                tableclass=table_class)
+
+        has_row_index = 'row_index' in self.colnames
```

Comment:
Or `index` ?
I don't understand why, if `has_row_index` is True, you add the column but don't remove it. Is it possible to add a Column with the same name as an existing one ?